reactor-netty-core/src/test/java/reactor/netty/transport/ProxyProviderTest.java

```diff
@@ -368,6 +369,42 @@ void proxyFromSystemProperties_customNonProxyHostsWithWildcardSetForHttpsProxy()
 		assertThat(provider.getNonProxyHostsPredicate().toString()).isEqualTo(".*\\Q.non-host\\E");
 	}
 
+	@Test
+	void proxyFromSystemProperties_basicAuthSetFromHttpProxy() {
+		Properties properties = new Properties();
+		properties.setProperty(ProxyProvider.HTTP_PROXY_HOST, "host");
+		properties.setProperty(ProxyProvider.HTTP_PROXY_USER, "user");
+		properties.setProperty(ProxyProvider.HTTP_PROXY_PASSWORD, "password");
+
+		ProxyProvider provider = ProxyProvider.createFrom(properties);
+		assertThat(provider).isNotNull();
+
+		ProxyHandler handler = provider.newProxyHandler();
+		assertThat(handler.getClass()).isEqualTo(HttpProxyHandler.class);
+
+		HttpProxyHandler httpHandler = (HttpProxyHandler) handler;
+		assertThat(httpHandler.username()).isEqualTo("user");
+		assertThat(httpHandler.password()).isEqualTo("password");
+	}
+
+	@Test
+	void proxyFromSystemProperties_basicAuthSetFromHttpsProxy() {
+		Properties properties = new Properties();
+		properties.setProperty(ProxyProvider.HTTPS_PROXY_HOST, "host");
+		properties.setProperty(ProxyProvider.HTTPS_PROXY_USER, "user");
+		properties.setProperty(ProxyProvider.HTTPS_PROXY_PASSWORD, "password");
+
+		ProxyProvider provider = ProxyProvider.createFrom(properties);
+		assertThat(provider).isNotNull();
+
+		ProxyHandler handler = provider.newProxyHandler();
+		assertThat(handler.getClass()).isEqualTo(HttpProxyHandler.class);
+
+		HttpProxyHandler httpHandler = (HttpProxyHandler) handler;
+		assertThat(httpHandler.username()).isEqualTo("user");
+		assertThat(httpHandler.password()).isEqualTo("password");
+	}
+
```

Comment:
add two tests to check the NPE reflects the property used (HTTP vs HTTPS) + at least one test to check what happens when `username(null)` is invoked?

Reply:
I added tests for the NPE cases. I don't think it's useful to add specific tests for when username is null, as it's already the case in all other tests.